### PR TITLE
Add support for Honeybadger

### PR DESCRIPTION
**Pull request by michaelherold**
Honeybadger is another exception service, like Rollbar. This change
introduces awareness of Ownership to Honeybadger. We [tag][1] each
notice with the current owner to allow you to filter by owner within
a single project.

Additionally, you can toggle between different API keys to route errors
to different projects to further segregate your errors for individual
teams. This requires configuration in the form:

```ruby
Ownership::Honeybadger.api_keys = {
  logistics: 'logistics-key',
  platform: 'platform-key'
}
```

If there is no separate key, we fall back to using the default key.

[1]: https://docs.honeybadger.io/lib/ruby/getting-started/tagging-errors.html

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -2,3 +2,8 @@ source "https://rubygems.org"
 
 # Specify your gem's dependencies in ownership.gemspec
 gemspec
+
+group :test do
```
Comment (ankane) [inline Gemfile]: Let's move this to gemspec as dev dependency.

File: test/honeybadger_test.rb [hunk 2]
```diff
@@ -0,0 +1,113 @@
+require_relative "test_helper"
+
+# Silence the boot up message from Honeybadger
+begin
+  original_stdout, $stdout = $stdout, StringIO.new
+  original_stderr, $stderr = $stderr, StringIO.new
+  require "honeybadger/ruby"
+  Honeybadger.init!(framework: :ruby, env: "test", :"logging.path" => "STDOUT")
```
Comment (ankane) [inline test/honeybadger_test.rb]: Can probably set `logging.path` to `Tempfile.new.path` so don't need to mess with stdout.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
+## Unreleased
+
+- Honeybadger integration
```
Comment (ankane) [inline CHANGELOG.md:5]: Thanks for changelog update - few people do this :)

File: lib/ownership/honeybadger.rb [hunk 4]
```diff
@@ -0,0 +1,48 @@
+module Ownership
+  module Honeybadger
+    class << self
+      attr_reader :api_keys
+
+      def api_keys=(api_keys)
+        @api_keys = api_keys
+        @configuration ||= configure
```
Comment (ankane) [inline lib/ownership/honeybadger.rb:8]: Thanks for following Rollbar convention

**Review by ankane — CHANGES_REQUESTED** [review]
Hey @michaelherold, overall, looks great. Thanks for including tests as well 💯 Added some feedback inline. I feel like we can probably leave out the context functionality to keep things simpler, since if needed, users can do:

```ruby
owner(:logistics) { Honeybadger.notify("Alert!!") }
```

File: CHANGELOG.md (hunk 3 above)
Comment (michaelherold) [reply CHANGELOG.md:5]: I maintain Hashie and it's a big pain for that gem so I always try to. 😄

**Comment by michaelherold** [comment]
Thanks for the great suggestions! I've implemented all of them in a separate commit with a merge commit to resolve the conflicts. I'm happy to rebase once you're good with it, but I thought it would be easiest to review like this.

**Comment by ankane** [comment]
Awesome, looks great. Squashed and merged.

Hashie is great, btw! I use it for [Searchkick](https://github.com/ankane/searchkick), another one of my projects.

**Pull request merged**